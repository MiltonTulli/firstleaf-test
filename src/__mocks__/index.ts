export const countries = [
  {
    name: 'Afghanistan',
    alpha2Code: 'AF',
    alpha3Code: 'AFG',
    capital: 'Kabul',
    region: 'Asia',
    subregion: 'Southern Asia',
    population: 27657145,
    latlng: [33, 65],
    demonym: 'Afghan',
    area: 652230,
    gini: 27.8,
    timezones: ['UTC+04:30'],
    borders: ['IRN', 'PAK', 'TKM', 'UZB', 'TJK', 'CHN'],
    nativeName: 'افغانستان',
    currencies: [
      {
        code: 'AFN',
        name: 'Afghan afghani',
        symbol: '؋'
      }
    ],
    languages: [
      {
        iso639_1: 'ps',
        iso639_2: 'pus',
        name: 'Pashto',
        nativeName: 'پښتو'
      },
      {
        iso639_1: 'uz',
        iso639_2: 'uzb',
        name: 'Uzbek',
        nativeName: 'Oʻzbek'
      },
      {
        iso639_1: 'tk',
        iso639_2: 'tuk',
        name: 'Turkmen',
        nativeName: 'Türkmen'
      }
    ],
    flag: 'https://restcountries.eu/data/afg.svg'
  },
  {
    name: 'Åland Islands',
    alpha2Code: 'AX',
    alpha3Code: 'ALA',
    capital: 'Mariehamn',
    region: 'Europe',
    subregion: 'Northern Europe',
    population: 28875,
    latlng: [60.116667, 19.9],
    demonym: 'Ålandish',
    area: 1580,
    gini: null,
    timezones: ['UTC+02:00'],
    borders: [],
    nativeName: 'Åland',
    currencies: [
      {
        code: 'EUR',
        name: 'Euro',
        symbol: '€'
      }
    ],
    languages: [
      {
        iso639_1: 'sv',
        iso639_2: 'swe',
        name: 'Swedish',
        nativeName: 'svenska'
      }
    ],
    flag: 'https://restcountries.eu/data/ala.svg'
  },
  {
    name: 'Albania',
    alpha2Code: 'AL',
    alpha3Code: 'ALB',
    capital: 'Tirana',
    region: 'Europe',
    subregion: 'Southern Europe',
    population: 2886026,
    latlng: [41, 20],
    demonym: 'Albanian',
    area: 28748,
    gini: 34.5,
    timezones: ['UTC+01:00'],
    borders: ['MNE', 'GRC', 'MKD', 'KOS'],
    nativeName: 'Shqipëria',
    currencies: [
      {
        code: 'ALL',
        name: 'Albanian lek',
        symbol: 'L'
      }
    ],
    languages: [
      {
        iso639_1: 'sq',
        iso639_2: 'sqi',
        name: 'Albanian',
        nativeName: 'Shqip'
      }
    ],
    flag: 'https://restcountries.eu/data/alb.svg'
  },
  {
    name: 'Algeria',
    alpha2Code: 'DZ',
    alpha3Code: 'DZA',
    capital: 'Algiers',
    region: 'Africa',
    subregion: 'Northern Africa',
    population: 40400000,
    latlng: [28, 3],
    demonym: 'Algerian',
    area: 2381741,
    gini: 35.3,
    timezones: ['UTC+01:00'],
    borders: ['TUN', 'LBY', 'NER', 'ESH', 'MRT', 'MLI', 'MAR'],
    nativeName: 'الجزائر',
    currencies: [
      {
        code: 'DZD',
        name: 'Algerian dinar',
        symbol: 'د.ج'
      }
    ],
    languages: [
      {
        iso639_1: 'ar',
        iso639_2: 'ara',
        name: 'Arabic',
        nativeName: 'العربية'
      }
    ],
    flag: 'https://restcountries.eu/data/dza.svg'
  }
];
