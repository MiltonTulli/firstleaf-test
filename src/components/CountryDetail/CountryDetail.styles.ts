import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  root: { padding: theme.spacing(3) },
  textContainer: {},
  labelValueItem: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    '& > *:nth-child(1)': {
      marginRight: 4,
      fontSize: 11
    },
    '& > *:nth-child(2)': {
      marginRight: 4,
      fontSize: 17
    }
  },
  map: {
    width: '100%',
    border: 0,
    height: 300
  }
}));
