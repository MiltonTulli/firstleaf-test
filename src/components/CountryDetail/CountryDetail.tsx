import React from 'react';
import { Grid, Button, Typography } from '@material-ui/core';
import ArrowBack from '@material-ui/icons/ArrowBack';
import { useHistory, RouteComponentProps, Redirect } from 'react-router-dom';
import _isEmpty from 'lodash/isEmpty';
import _get from 'lodash/get';
import { Country } from '../../interfaces';
import R from '../../utils/routes';
import env from '../../config';
import useStyles from './CountryDetail.styles';

interface ICountryDetailProps
  extends RouteComponentProps<
    {},
    any,
    {
      country: Country;
    }
  > {
  countries: Country[];
}

export const CountryDetail: React.FunctionComponent<ICountryDetailProps> = (
  props: ICountryDetailProps
) => {
  const classes = useStyles();
  const history = useHistory();

  const country: Country = _get(props, 'location.state.country', {});

  if (_isEmpty(country)) return <Redirect to={R.APP} />;

  const latX = _get(country, 'latlng[0]', 0);
  const latY = _get(country, 'latlng[1]', 0);
  const mapSrc = `https://www.google.com/maps/embed/v1/place?q=${latX},${latY}&key=${env.googleApiKey}&zoom=4`;

  return (
    <div className={classes.root}>
      <Button variant="outlined" onClick={() => history.push(R.APP)}>
        <ArrowBack /> back
      </Button>
      <Grid container spacing={2} alignItems="center">
        <Grid item xs={12} sm={6} className={classes.textContainer}>
          <Grid container spacing={2} justify="center" alignItems="center">
            <Grid item xs={12}>
              <Grid container spacing={1} justify="center" alignItems="center">
                <Grid item>
                  <Typography variant="body2">[{country.alpha3Code}]</Typography>
                </Grid>
                <Grid item>
                  <Typography data-testid="name" variant="h4">
                    {country.name}
                  </Typography>
                </Grid>
                <Grid item>
                  <img
                    alt={`${country.name} flag`}
                    src={country.flag}
                    style={{ height: 30, width: 40 }}
                  ></img>
                </Grid>
              </Grid>
            </Grid>
            <Grid item xs={12} className={classes.labelValueItem}>
              <Typography align="center" variant="subtitle2">
                Capital:
              </Typography>
              <Typography data-testid="capital" align="center" variant="body2">
                {country.capital}
              </Typography>
            </Grid>
            <Grid item xs={12} className={classes.labelValueItem}>
              <Typography align="center" variant="subtitle2">
                Currencies:
              </Typography>
              <Typography align="center" variant="body2">
                {country.currencies.map(c => c.symbol).join(' - ')}
              </Typography>
            </Grid>
            <Grid item xs={12} className={classes.labelValueItem}>
              <Typography align="center" variant="subtitle2">
                Region:
              </Typography>
              <Typography align="center" variant="body2">
                {country.region}
              </Typography>
            </Grid>
            <Grid item xs={12} className={classes.labelValueItem}>
              <Typography align="center" variant="subtitle2">
                Population:
              </Typography>
              <Typography align="center" variant="body2">
                {country.population}
              </Typography>
            </Grid>
            <Grid item xs={12} className={classes.labelValueItem}>
              <Typography align="center" variant="subtitle2">
                Subregion:
              </Typography>
              <Typography align="center" variant="body2">
                {country.subregion}
              </Typography>
            </Grid>
            <Grid item xs={12} className={classes.labelValueItem}>
              <Typography align="center" variant="subtitle2">
                Timezone/s:
              </Typography>
              <Typography align="center" variant="body2">
                {country.timezones.map((c: string) => c).join(' - ')}
              </Typography>
            </Grid>
          </Grid>
        </Grid>

        <Grid item xs={12} sm={6}>
          <iframe title={`${country.name} map`} className={classes.map} src={mapSrc}></iframe>
        </Grid>
      </Grid>
    </div>
  );
};
