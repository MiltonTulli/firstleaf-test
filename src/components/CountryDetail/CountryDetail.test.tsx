import React from 'react';
import { shallow } from 'enzyme';
import { countries } from '../../__mocks__';
import { CountryDetail } from './CountryDetail';

describe('<CountryDetail>', () => {
  let props;
  beforeEach(() => {
    props = {
      history: {},
      location: {
        state: {
          country: countries[0]
        }
      },
      match: {}
    };
  });
  it('Should match snapshot', () => {
    const wrapper = shallow(<CountryDetail {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Should show country name', () => {
    const wrapper = shallow(<CountryDetail {...props} />);
    const node = wrapper.find('[data-testid="name"]');
    expect(node.text()).toEqual(props.location.state.country.name);
  });
  it('Should show country capital', () => {
    const wrapper = shallow(<CountryDetail {...props} />);
    const node = wrapper.find('[data-testid="capital"]');
    expect(node.text()).toEqual(props.location.state.country.capital);
  });
});
