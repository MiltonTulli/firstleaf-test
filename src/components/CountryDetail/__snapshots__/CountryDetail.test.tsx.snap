// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryDetail> Should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    variant="outlined"
  >
    <Memo(ForwardRef(ArrowBackIcon)) />
     back
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-textContainer-2"
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justify="center"
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Grid))
            alignItems="center"
            container={true}
            justify="center"
            spacing={1}
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                variant="body2"
              >
                [
                AFG
                ]
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                data-testid="name"
                variant="h4"
              >
                Afghanistan
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <img
                alt="Afghanistan flag"
                src="https://restcountries.eu/data/afg.svg"
                style={
                  Object {
                    "height": 30,
                    "width": 40,
                  }
                }
              />
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-labelValueItem-3"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="subtitle2"
          >
            Capital:
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            align="center"
            data-testid="capital"
            variant="body2"
          >
            Kabul
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-labelValueItem-3"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="subtitle2"
          >
            Currencies:
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="body2"
          >
            Ø‹
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-labelValueItem-3"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="subtitle2"
          >
            Region:
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="body2"
          >
            Asia
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-labelValueItem-3"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="subtitle2"
          >
            Population:
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="body2"
          >
            27657145
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-labelValueItem-3"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="subtitle2"
          >
            Subregion:
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="body2"
          >
            Southern Asia
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-labelValueItem-3"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="subtitle2"
          >
            Timezone/s:
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            align="center"
            variant="body2"
          >
            UTC+04:30
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <iframe
        className="makeStyles-map-4"
        src="https://www.google.com/maps/embed/v1/place?q=33,65&key=AIzaSyBx0MPM-levclO7Hl5NHSczvHUb6gLk0b4&zoom=4"
        title="Afghanistan map"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
